### PR TITLE
fix(#7): re-enable z-axis offset between layers and provide a callback to attach additional components to colliders

### LINKED ISSUE #7: Z-order of tile/layer affects collisions
I yanked 0.3.2 because of an issue that was caused by the z-order changes.

The colliders also use the transform of the tile, and that is nested below the transform of the layer, which is nested below the map itself.

We probably need a better approach for this.
For now I will revert the z-offset change, with a comment. And leave this issue open for a more comprehensive fix.
It's likely that we need to somehow make the z-offset of the rendering layer independent of the physics. I'm not sure how to achieve that yet.

The goals of this ticket are:
1. Make tilemap layers respect the z coordinate for rendering
2. Make the physics still work correctly. Part of this is understanding what rapier does with the z-coordinate even in 2D. This is confusing to me. It doesn't seem to be documented and may be worth a GH issue.

### PR DESCRIPTION
**Pull request by adrien-bon**
Fixes #7

### CODE REVIEW

File: src/physics/rapier/shapes.rs [hunk 1]
```diff
@@ -134,7 +134,7 @@ pub(crate) fn insert_colliders_from_shapes(
         } * Transform::from_translation(Vec3::new(pos.x, pos.y, 0.));
 
         commands
-            .spawn((collider, TransformBundle::from_transform(transform)))
+            .spawn((RigidBody::KinematicPositionBased, collider, TransformBundle::from_transform(transform)))
```
Comment (stevepryde) [inline src/physics/rapier/shapes.rs:137]: I feel like the default should be `Fixed` ?

Comment (stevepryde) [reply src/physics/rapier/shapes.rs:137]: Although the example might require kinematic, so you'd need this to be configurable in order to make the example work...

Comment (stevepryde) [reply src/physics/rapier/shapes.rs:137]: Ironically my game works better without any rigid body here :thinking:. I'm still confused by that. I get the feeling that rotating the entire map like I was doing is not well supported. The movements at the edges of the map are probably too large for the physics to work smoothly

**Comment by adrien-bon** [comment]
Added a callback system so the user can set up his own logic. By default, we only add the collider (same behaviour as before).
For the example, I sticked with the behaviour that worked best for me, ie. a `RigidBody::KinematicPositionBased` for the tiles and a `RigidBody::Dynamic` for the player.

**Pull request merged**